habit-backend/users/views.py

```diff
@@ -0,0 +1,89 @@
+import re
+
+from django.contrib.auth import authenticate, login
+from rest_framework import permissions, generics, status
+from rest_framework.response import Response
+from rest_framework_jwt.settings import api_settings
+from .userhelper import ValidateEmail
+
+from users.models import User
+from users.serializers import UserSerializer , LoginSerializer
+
+jwt_payload_handler = api_settings.JWT_PAYLOAD_HANDLER
+jwt_encode_handler = api_settings.JWT_ENCODE_HANDLER
+
+
+class SignUpUserView(generics.GenericAPIView):
+    permission_classes = (permissions.AllowAny,)
+
+    def post(self, request, *args, **kwargs):
+
+        email = request.data.get('email')
+        if email is None:
+            return Response({'email': ['This field is required.']},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        email = email.lower()
+        email_valid_status = ValidateEmail(email)
+        if email_valid_status == "Invalid":
```

Comment:
If we return True/False from ValidateEmail, we can straightaway check for ValidateEmail(email) and do away with the `email_valid_status` variable. It also saves us from using hard-coded strings